packages/ipfs-core/src/components/bootstrap/add.js

```diff
@@ -30,11 +31,13 @@ module.exports = ({ repo }) => {
 
     const config = await repo.config.getAll(options)
 
+    // @ts-ignore - May not have `Bootstrap`
     if (config.Bootstrap.indexOf(multiaddr.toString()) === -1) {
+      // @ts-ignore - May not have `Bootstrap`
       config.Bootstrap.push(multiaddr.toString())
     }
 
-    await repo.config.set(config)
+    await repo.config.replace(config)
```

Comment:
Because interface of `set` is `set(path,:string value:any)` and replace has `replace(value:Config)`. Later can ensure that what you pass is what's expected former can not. I also did not wanted to do the `set` overload typing because they tend to have issues.